### DISCUSSION TITLE
Remove avd-ksv-0011 (CPU limit recommendation)

### DISCUSSION THREAD

**Discussion by jemag** [aquasecurity/trivy#5045] [Ideas]
### Description

Currently trivy will recommend adding a CPU limit when it is missing according to https://avd.aquasec.com/misconfig/kubernetes/general/avd-ksv-0011/.

In the last few years, the community has mostly moved to **no** CPU limit being the better default in the vast majority of cases. I think giving this warning might mislead users into adding CPU limits when they really should not. This [article](https://home.robusta.dev/blog/stop-using-cpu-limits) mentions some of the reasons behind this.

### Target

Kubernetes

### Scanner

Misconfiguration

**Comment by simar7** [comment]
@chen-keinan do you have any recommendation on this?

**Reply by chen-keinan** [reply]
@jemag @simar7 I'm not sure about removing it as it look like controversial matter , see other [opinion](https://komodor.com/learn/kubernetes-cpu-limits-throttling/)  this rule has `Low` severity, meaning its up for user to take decision on his free time after investigating its not marked as urgent.

**Reply by jemag** [reply]
@chen-keinan unfortunately the problems listed in the https://komodor.com/learn/kubernetes-cpu-limits-throttling/ article are related to improperly configured CPU requests, not limits.

CPU request guarantees CPU available to the container and containers should be able to run properly within their CPU requests. That means that other containers bursting up to the node capacity will not impact their operation. CPU starvation is therefore an non-issue with properly configured CPU requests. The same can be said for Kubernetes components (for example, clusters like AKS will reserve resources for the core components on each node).

I would say that for probably 99% of generic use cases, CPU limit are more of a detriment, so even if the recommendation is `low` it might mislead a large portion of users that do not know better.

I think removing is the better option, but if not possible, I would suggest at least clarifying the current description so we do not mislead users into adding it when it is to their detriment.

**Reply by chen-keinan** [reply]
@jemag clarifying the description will be reasonable.
would you like to raise a PR?  here is [ref point](https://github.com/aquasecurity/defsec/blob/ae6078ba0df9c43c9f10ff4d992f0eb4180c155d/rules/kubernetes/policies/general/CPU_not_limited.rego#L11)
